rest-api/dao/participant_dao.py

```diff
@@ -339,6 +348,13 @@ def add_missing_hpo_from_site(self, session, participant_id, site_id):
     # Update the version and add history row
     self._do_update(session, participant, participant)
 
+  def handle_integrity_error(self, tried_ids, e, obj):
+    if 'external_id' in e.message:
+      existing_participant = self._check_if_external_id_exists(obj)
+      if existing_participant:
+        return existing_participant
+    return logging.warning('Failed insert with %s: %s', tried_ids, e.message)
```

Comment:
This should be:

return super(self, ParticipantDao).handle_integrity_error(tried_ids, e, obj)

Reply:
ha, so I had that at first and thought I'd created some sort of circular reference or something, totally forgetting about the max_retries. Anyway, done.